src/Matterhorn/State/Messages.hs

```diff
@@ -839,17 +861,39 @@ notifyGetPayload NotifyV3 st post mentioned = do
         where
             msg = sanitizeUserText $ postMessage post
             sender = maybePostUsername st post
-            msgTy = case postType post of
-                PostTypeJoinChannel       -> "joinChannel"
-                PostTypeLeaveChannel      -> "leaveChannel"
-                PostTypeAddToChannel      -> "addToChannel"
-                PostTypeRemoveFromChannel -> "removeFromChannel"
-                PostTypeHeaderChange      -> "headerChange"
-                PostTypeDisplayNameChange -> "displayNameChange"
-                PostTypePurposeChange     -> "purposeChange"
-                PostTypeChannelDeleted    -> "channelDeleted"
-                PostTypeEphemeral         -> "ephemeral"
-                PostTypeUnknown _         -> "unknown"
+            msgTy = notifyGetMessageType post
+notifyGetPayload NotifyV4 st post mentioned = do
+    chan <- findChannelById (post^.postChannelIdL) (st^.csChannels)
+    let chName = chan^.ccInfo.cdName
+    let chType = notifyGetChannelType chan
```

Comment:
I'm not 100% sure if this is the right approach, but if I let them in the `where` context then the `chan` variable was out of scope there.

Reply:
Yeah, that ends up being necessary for exactly the reason you found. I would only recommend one minor style change: `let`s can be batched like so:
```haskell
let foo = ...
    bar = ...
```
instead of one `let` per binding.

Reply:
Ah, neat, thanks!